### DISCUSSION TITLE
如何更新ipv6.txt地址段获得更优v6 IP？

### DISCUSSION THREAD

**Discussion by dyinandy** [XIU2/CloudflareSpeedTest#131] [提问]
首先，非常感谢@XIU2提供如此好用的工具，最近弄IPV6，不过看到最新版的v1.5.1似乎只有：2606:4700::/32 内的v6地址。
Cloudflare 官网的的ipv6地址池数量太大的，在哪里可以查看到那些IP是available的。多谢。

[Cloudflare IP Ranges](https://www.cloudflare.com/ips/)
2400:cb00::/32
**2606:4700::/32**
2803:f800::/32
2405:b500::/32
2405:8100::/32
2a06:98c0::/29
2c0f:f248::/32

**Comment by XIU2** [comment]
Cloudflare 只公布了其所有 IP 段，并没有说明哪些 IP 段目前在用、各个 IP 段分别是用来干嘛的。

我只是自己简单扫了扫，发现所有可用的 IPv6 都集中在 `2606:4700::/32` 这一个 IP 段中 #120 。

****

用我这个工具对每个 IP 段单独扫描 20 次，除了 `2606:4700::/32` 外，其他 IP 段都**没有找到任何一个可用**的 IPv6 地址，所以结果很明显了，要么**这些 IP 段没有启用**，要么就是这些 IP 段**只有极少极少的 IP 在使用中**。
然后我再对  `2606:4700::/32` 根 IP 段进行细扫，得到了更准确的子 IP 段。

再加上扫描各个使用 Cloudflare CDN 的**域名 AAAA 解析记录**（IPv6 地址）。

两者结合后综合判断才得到了现在的 `ipv6.txt` 文件。   
> 虽然已经有更细致、准确的子 IP 段了，但是为了一定程度上避免遗漏，我又把 `2606:4700::/32` 根 IP 段加到 ipv6.txt 中，就当碰运气了~

**Reply by dyinandy** [reply]
明白了，原来发布包里的ipv6.txt是这样得到了，辛苦了！谢谢。
